### PR TITLE
HostDisplay: Create swap chain in CreateRenderDevice() for D3D

### LINKED ISSUE #6222: [BUG]: Failed to acquire exclusive fullscreen
### Describe the Bug

Using Command Line when "Fullscreen Mode" is not set to "Borderless Fulscreen", game is not launched in fullscreen and the following error occurs: "Failed to acquire exclusive fullscreen". But if the "Fullscreen Mode" setting is set to "Borderless Fullscreen", the game launch in fullscreen normally 

### Reproduction Steps

pcsx2-qtx64-avx2.exe -fullscreen -- "PathToGame"

### Expected Behavior

Game starts in exclusive fullscreen mode

### PCSX2 Revision

v1.7.2816

### Operating System

Windows 11

### If Linux - Specify Distro

_No response_

### PR DESCRIPTION
**Pull request by stenzek**
### Description of Changes

Fixes starting in exclusive fullscreen mode in Qt.

### Rationale behind Changes

Fixes #6222 .

### Suggested Testing Steps

Test starting with exclusive fullscreen, ensure mode switch succeeds.

### CODE REVIEW

**Review by JordanTheToaster — APPROVED** [review]

File: pcsx2/Frontend/D3D11HostDisplay.cpp [hunk 1]
```diff
@@ -320,14 +320,15 @@ bool D3D11HostDisplay::CreateRenderDevice(const WindowInfo& wi, std::string_view
...
 	return true;
 }
 
 bool D3D11HostDisplay::InitializeRenderDevice(std::string_view shader_cache_directory, bool debug_device)
 {
-	if (m_window_info.type != WindowInfo::Type::Surfaceless && !CreateSwapChain(nullptr))
-		return false;
-
 	return true;
 }
```
Comment (refractionpcsx2) [inline pcsx2/Frontend/D3D11HostDisplay.cpp:333]: Is there any point in this function existing anymore? lol

File: pcsx2/Frontend/D3D12HostDisplay.cpp [hunk 2]
```diff
@@ -191,14 +191,15 @@ bool D3D12HostDisplay::CreateRenderDevice(const WindowInfo& wi, std::string_view
...
 	return true;
 }
 
 bool D3D12HostDisplay::InitializeRenderDevice(std::string_view shader_cache_directory, bool debug_device)
 {
-	if (m_window_info.type != WindowInfo::Type::Surfaceless && !CreateSwapChain(nullptr))
-		return false;
-
 	return true;
 }
```
Comment (refractionpcsx2) [inline pcsx2/Frontend/D3D12HostDisplay.cpp:204]: same with this one

File: pcsx2/Frontend/D3D11HostDisplay.cpp (hunk 1 above)
Comment (refractionpcsx2) [reply pcsx2/Frontend/D3D11HostDisplay.cpp:333]: Okay apparently required as it's virtual for GL

File: pcsx2/Frontend/D3D12HostDisplay.cpp (hunk 2 above)
Comment (refractionpcsx2) [reply pcsx2/Frontend/D3D12HostDisplay.cpp:204]: Okay apparently required as it's virtual for GL

**Pull request merged**